tests/playwright-test/golden.spec.ts

```diff
@@ -96,6 +176,52 @@ test('should update expectations', async ({runInlineTest}, testInfo) => {
   expect(data.toString()).toBe('Hello world updated');
 });
 
+test('should update expectations with negated matcher', async ({runInlineTest}, testInfo) => {
+  const result = await runInlineTest({
+    'a.spec.js-snapshots/snapshot.txt': `Hello world`,
+    'a.spec.js': `
+      const { test } = pwt;
+      test('is a test', ({}) => {
+        expect('Hello world updated').not.toMatchSnapshot('snapshot.txt');
+      });
+    `
+  }, { 'update-snapshots': true });
+  expect(result.exitCode).toBe(0);
+  expect(result.output).toContain('snapshot.txt does not match, writing actual.');
+  const data = fs.readFileSync(testInfo.outputPath('a.spec.js-snapshots/snapshot.txt'));
+  expect(data.toString()).toBe('Hello world updated');
+});
+
+test('should silently write missing expectations locally with the update-snapshots flag', async ({runInlineTest}, testInfo) => {
```

Comment:
Don't we already have a test for this?

Reply:
It's a different one. The test you're talking about assumed that a snapshot already exists and we need to update it. This one assumes that snapshot doesn't exist and we need to create a new one.

Reply:
Although I'm not really sure about the `locally` part where we pass an empty `CI` env variable. I copied it from another test and probably it shouldn't be here. If you have some knowledge about it, I'd be glad to hear it out.